engine/chassis.ts

```diff
@@ -338,14 +341,14 @@ export class Program {
             const textures = samplerTextures?.filter((texture) => texture.set === descriptorSetId) || [];
 
             TestBase.assert(blocks.length < TEST_UBO_COUNTS[descriptorSetId] || TEST_UBO_COUNTS[descriptorSetId] === -1,
-                `too many UBOs in set ${descriptorSetId}, only ${TEST_UBO_COUNTS[descriptorSetId]} seats exist.`);
-            TestBase.assert(blocks.length < 8 && TEST_UBO_COUNTS[descriptorSetId] !== -1,
-                `too many UBOs in set ${descriptorSetId}, only ${TEST_UBO_COUNTS[descriptorSetId]} seats exist.`);
+                `too many UBOs in set ${descriptorSetId}, only ${TEST_UBO_COUNTS[descriptorSetId]} slots exist.`);
+            TestBase.assert(blocks.length < TEST_VARY_SLOT_LIM || TEST_UBO_COUNTS[descriptorSetId] !== -1,
+                `too many UBOs in set ${descriptorSetId}, only ${TEST_UBO_COUNTS[descriptorSetId]} slots exist.`);
 
             TestBase.assert(textures.length < TEST_SAMPLER_COUNTS[descriptorSetId] || TEST_SAMPLER_COUNTS[descriptorSetId] === -1,
-                `too many UBOs in set ${descriptorSetId}, only ${TEST_SAMPLER_COUNTS[descriptorSetId]} seats exist.`);
-            TestBase.assert(textures.length < 8 && TEST_SAMPLER_COUNTS[descriptorSetId] !== -1,
-                `too many UBOs in set ${descriptorSetId}, only ${TEST_SAMPLER_COUNTS[descriptorSetId]} seats exist.`);
+                `too many Textures in set ${descriptorSetId}, only ${TEST_SAMPLER_COUNTS[descriptorSetId]} slots exist.`);
+            TestBase.assert(textures.length < TEST_VARY_SLOT_LIM || TEST_SAMPLER_COUNTS[descriptorSetId] !== -1,
```

Comment:
Textures limit is a completely different thing than the UBO limit, and should be calculated separately